arrow-libs/core/arrow-core/src/commonMain/kotlin/arrow/core/computations/result.kt

```diff
@@ -0,0 +1,11 @@
+package arrow.core.computations
+
+public object ResultEffect {
+  public fun <A> Result<A>.bind(): A = getOrThrow()
```

Comment:
Would it be in the spirit of this Effect to have cross bind like:

```
fun <A> A?.bindOr(f: () -> Throwable): A = this ?: throw f()
```

Happy to do this on a separate PR :slightly\_smiling\_face:

Reply:
Actually, there is no need for this. You can just throw inside `result { }` since it's just a simple DSL on top of `runCatching`.

Reply:
Very true, we can do:

```
val profile = nullableProfile ?: error("MissingProfile")
```

or

```
val profile = checkNotNull(nullableProfile)
```

Maybe we should add this to the documentation :thinking: